vllm/v1/worker/gpu_model_runner.py

```diff
@@ -509,8 +512,14 @@ def execute_model(
             logprobs = None
         else:
             logprobs = sampler_output.logprobs.cpu()
+
+        assert all(
+            req_id is not None for req_id in
+            self.input_batch.req_ids[:num_reqs]), "req_ids contains None"
+        req_ids = cast(List[str], self.input_batch.req_ids[:num_reqs])
```

Comment:
QQ: why is this line needed?

Reply:
This is the error:

```
vllm/v1/worker/gpu_model_runner.py:546: error: Argument "req_ids" to "ModelRunnerOutput" has incompatible type "list[Optional[str]]"; expected "list[str]"  [arg-type]
```

`input_batch.req_ids` can have `None` elements:

```
self.req_ids: List[Optional[str]] = [None] * max_num_reqs
```

but `ModelRunnerOutput.req_ids` can't:

```
class ModelRunnerOutput:

    # [num_reqs]                                                                                                                                                                                                                                                                  
    req_ids: List[str]
```

AFAIR `input_batch.num_reqs` tells us how many are non-None?

Reply:
Got it. Thanks!